shopfloor/services/checkout.py

```diff
@@ -1315,8 +1333,28 @@ def done(self, picking_id, confirmation=False):
                         "body": _("Remaining raw product not packed, proceed anyway?"),
                     },
                 )
-        stock = self._actions_for("stock")
         lines_done = self._lines_checkout_done(picking)
+        dest_location = picking.location_dest_id
+        child_locations = self.env["stock.location"].search(
+            [("id", "child_of", dest_location.id), ("usage", "!=", "view")]
+        )
+        if selected_child_location_id:
+            for line in lines_done:
+                line.update({"location_dest_id": selected_child_location_id})
+        elif len(child_locations) > 0 and child_locations != dest_location:
+            return self._response_for_select_child_location(
+                picking,
+                child_locations,
+                message={
+                    "message_type": "warning",
+                    "body": _(
+                        "Select the destination location inside {}".format(
```

Comment:
Wrong interpolation of translated string. Also, it would be nice to move this msg to the message registry.